File: coko/com.elster.jupiter.issue.task.extjs/src/main/web/js/dal/src/model/CreationRuleActionPhase.js
/*
 * Copyright (c) 2017 by Honeywell International Inc. All Rights Reserved
 */

Ext.define('Itk.model.CreationRuleActionPhase', {
    extend: 'Ext.data.Model',
    fields: [
        {
            name: 'uuid',
            type: 'string'
        },
        {
            name: 'title',
            type: 'string'
        },
        {
            name: 'description',
            type: 'string'
        }
    ],

    idProperty: 'uuid'
});

File: comu/com.energyict.mdc.device.alarms/src/main/java/com/energyict/mdc/device/alarms/entity/CreationRuleActionPhase.java
/*
 * Copyright (c) 2017 by Honeywell International Inc. All Rights Reserved
 */

package com.energyict.mdc.device.alarms.entity;

public enum CreationRuleActionPhase {
    CREATE("DeviceAlarmActionPhaseCreation", "DeviceAlarmActionPhaseCreationDescription"),
    OVERDUE("DeviceAlarmActionPhaseOverdue", "DeviceAlarmActionPhaseOverdueDescription");

    private String titleId;
    private String descriptionId;

    CreationRuleActionPhase(String titleId, String descriptionId) {
        this.titleId = titleId;
        this.descriptionId = descriptionId;
    }

    public String getTitleId(){
        return titleId;
    }

    public String getDescriptionId(){
        return descriptionId;
    }

    public static CreationRuleActionPhase fromString(String phase) {
        for (CreationRuleActionPhase column : CreationRuleActionPhase.values()) {
            if (column.name().equalsIgnoreCase(phase)) {
                return column;
            }
        }
        return null;
    }
}

